### PR TITLE
Core rearchitecture for modularity

### PR DESCRIPTION
**Pull request by mcharleb**
Layout was modified to be consistent and enable platform support to be separated into git repos and to add vendor collections of build targets.

The New vendor layout allows external git repos to be mapped in under vendor/ and will be automatically integrated into the PX4 build platform.
```
vendor/
└── ATLFlight
    ├── cmake
    ├── CMakeLists.txt
    ├── multi.conf
    ├── posix
    │   ├── cmake
    │   │   ├── posix_sdflight_default.cmake
    │   │   └── posix_sdflight_legacy.cmake
    │   ├── eagle
    │   │   ├── board_config.h
    │   │   ├── config_default.cmake
    │   └── excelsior
    │       ├── board_config.h
    │       ├── config_default.cmake
    │       └── config_legacy.cmake
    └── qurt
        ├── cmake
        │   ├── qurt_sdflight_default.cmake
        │   └── qurt_sdflight_legacy.cmake
        ├── eagle
        │   ├── board_config.h
        │   ├── config_default.cmake
        │   ├── config_legacy.cmake
        │   ├── config_muorb.cmake
        │   ├── config_travis.cmake
        │   └── config_test.cmake
        └── excelsior
            ├── board_config.h
            ├── config_default.cmake
            └── config_legacy.cmake
```
The existing boards and build targets have been reorganized:
```
platforms/posix/boards/
├── bebop
│   ├── board_config.h
│   └── config_default.cmake
├── ocpoc
│   ├── board_config.h
│   ├── config_cross.cmake
│   └── config_ubuntu.cmake
├── rpi
│   ├── board_config.h
│   ├── config_cross.cmake
│   ├── config_cross-no-shield.cmake
│   ├── config_native.cmake
│   ├── posix_rpi_common.cmake
│   └── posix_rpi_cross.cmake
└── sitl
    ├── board_config.h
    ├── config_broadcast.cmake
    ├── config_default.cmake
    ├── config_ekf2.cmake
    ├── config_inav.cmake
    ├── config_lpe.cmake
    ├── config_replay.cmake
    ├── config_shell.cmake
    └── config_test.cmake
```
The px4 prefixed targets are handled specially
```
platforms/nuttx/boards/px4/
├── cannode-v1
│   ├── board_config.h
│   ├── config_bootloader.cmake
│   └── config_default.cmake
├── esc-v1
│   ├── board_config.h
│   ├── config_bootloader.cmake
│   └── config_default.cmake
├── flow-v2
│   ├── board_config.h
│   └── config_bootloader.cmake
├── fmu-v1
│   ├── board_config.h
│   └── config_default.cmake
├── fmu-v2
│   ├── board_config.h
│   ├── config_default.cmake
│   ├── config_lpe.cmake
│   └── config_test.cmake
├── fmu-v3
│   ├── board_config.h
│   └── config_default.cmake
├── fmu-v4
│   ├── board_config.h
│   └── config_default.cmake
├── fmu-v4pro
│   ├── board_config.h
│   └── config_default.cmake
├── fmu-v5
│   ├── board_config.h
│   └── config_default.cmake
├── nucleoF767ZI-v1
│   ├── board_config.h
│   └── config_default.cmake
└── stm32f4discovery
    ├── board_config.h
    └── config_default.cmake
```

Other boards use a different layout:
```
platforms/nuttx/boards
├── aerocore2
│   └── config_default.cmake
├── aerofc-v1
│   └── config_default.cmake
├── auav-x21
│   ├── board_config.h
│   └── config_default.cmake
├── crazyflie
│   ├── board_config.h
│   └── config_default.cmake
├── esc35-v1
│   ├── board_config.h
│   ├── config_bootloader.cmake
│   └── config_default.cmake
├── mindpx-v2
│   ├── board_config.h
│   └── config_default.cmake
├── s2740vc-v1
│   ├── board_config.h
│   ├── config_bootloader.cmake
│   └── config_default.cmake
├── tap-v1
│   ├── board_config.h
│   └── config_default.cmake
└── zubaxgnss-v1
    ├── board_config.h
    └── config_bootloader.cmake

```

### CODE REVIEW

**Comment by mcharleb** [comment]
@dagar @davids5 I think the layout is as we discussed. File completion works, board_config.h was moved for all targets. ROMSFS is unchanged, and rc.sensors was not created per board.

File: src/lib/version/version.c [hunk 1]
```diff
@@ -206,13 +204,10 @@ const char *px4_firmware_git_branch(void)
 	return PX4_GIT_BRANCH_NAME;
 }
 
+#if !defined(__PX4_NUTTX)
```
Comment (davids5) [inline src/lib/version/version.c]: @mcharleb Is this an API change?

**Comment by dagar** [comment]
Great, I'll give it a try later today.

File: src/lib/version/version.c (hunk 1 above)
Comment (mcharleb) [inline src/lib/version/version.c]: This whole block of code is implemented in a new file specifically for the Nuttx build. The API is unchanged.

Comment (mcharleb) [inline src/lib/version/version.c]: platforms/nuttx/src/lib/version/version_nuttx.c

**Comment by mcharleb** [comment]
@LorenzMeier All the initial changes planned have been made AFAIK.

**Comment by LorenzMeier** [comment]
@mcharleb Is the proposal to move the PX4 boards eventually also into a "PX4 vendor" or do you feel a different layout is appropriate?

**Comment by davids5** [comment]
@mcharleb I pulled an built this.

Params seem to be messed up, I ran master on the same HW after and it was fine.

```
[boot] Fault Log info File No 4 Length 3177 flags:0x01 state:1
[boot] Fault Log is Armed
sercon: Registering CDC/ACM serial driver
sercon: Successfully registered the CDC/ACM serial driver
[i] microSD mounted: /fs/microsd
[hardfault_log] Fault Log is Armed
INFO  [param] selected parameter default file /fs/mtd_params
rgbled on I2C bus 2 at 0x55 (bus: 100 KHz, max: 100 KHz)
ERROR [param] Parameter SYS_AUTOCONFIG not found
ERROR [param] Parameter BAT_N_CELLS not found
ERROR [param] Parameter SYS_HITL not found
ERROR [param] Parameter SYS_USE_IO not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
ERROR [param] Parameter SYS_AUTOSTART not found
INFO  [dataman] Unknown restart, data manager file '/fs/microsd/dataman' size is 350224 bytes
MS5611_SPI on SPI bus 4 at 14 (20000 KHz)
WARN  [ms5611] no device on bus 4
MS5611_SPI on SPI bus 1 at 3 (20000 KHz)
WARN  [bst] no devices found
INFO  [ver] match: PX4FMU_V2
HMC5883_I2C on I2C bus 1 at 0x1e (bus: 100 KHz, max: 400 KHz)
WARN  [lis3mdl] no device on bus 2
WARN  [hmc5883] no device on bus 2 (type: 1)
WARN  [mpu6000] no device on bus 3
MPU6000 on SPI bus 1 at 4 (1000 KHz)
WARN  [mpu6000] no device on bus 5
WARN  [mpu9250] probe failed! 104
WARN  [mpu9250] no device on bus 3
WARN  [mpu9250] probe failed! 255
WARN  [mpu9250] no device on bus 4
L3GD20 on SPI bus 1 at 1 (11000 KHz)
LSM303D on SPI bus 1 at 2 (11000 KHz)
ERROR [sdp3x_airspeed] reset failed
WARN  [sdp3x_airspeed] trying SDP3X 2
ERROR [sdp3x_airspeed] reset failed
WARN  [sdp3x_airspeed] SDP3X init fail
WARN  [sdp3x_airspeed] no SDP3X airspeed sensor connected
ERROR [sdp3x_airspeed] reset failed
WARN  [sdp3x_airspeed] trying SDP3X 2
ERROR [sdp3x_airspeed] reset failed
WARN  [sdp3x_airspeed] SDP3X init fail
WARN  [sdp3x_airspeed] no SDP3X airspeed sensor connected
WARN  [ms5525_airspeed] trying MS5525 address 2
WARN  [ms5525_airspeed] MS5525 init fail
WARN  [ms5525_airspeed] no MS5525 airspeed sensor connected
Airspeed on I2C bus 1 at 0x28 (bus: 100 KHz, max: 100 KHz)
ERROR [ms4525_airspeed] already started
WARN  [ets_airspeed] no ETS airspeed sensor connected
WARN  [ets_airspeed] no ETS airspeed sensor connected
ERROR [lib__controllib] error finding param: BAT_V_EMPTY
ERROR [lib__controllib] error finding param: BAT_V_CHARGED
ERROR [lib__controllib] error finding param: BAT_N_CELLS
ERROR [lib__controllib] error finding param: BAT_CAPACITY
ERROR [lib__controllib] error finding param: BAT_V_LOAD_DROP
ERROR [lib__controllib] error finding param: BAT_R_INTERNAL
ERROR [lib__controllib] error finding param: BAT_LOW_THR
ERROR [lib__controllib] error finding param: BAT_CRIT_THR
ERROR [lib__controllib] error fiINFO  [mavlink] mode: Normal, danding param: BAT_EMERGEN_THR
ta rate: 1200 B/s on /dev/ttyS1 @ 57600B
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
ERROR [param] Parameter SYS_COMPANION not found
INFO  [ver] match: PX4FMU_V2
px4flow [210:100]
WARN  [px4flow] scanning I2C busINFO  [mavlink] mode: Config, daes for device..
ta rate: 800000 B/s on /dev/ttyACM0 @ 57600B
ERROR [param] Parameter SYS_LOGGER not found
INFO  [logger] logger started (mode=all)
No autostart ID found
ERROR [param] Parameter SYS_CAL_GYRO not found
ERROR [param] Parameter SYS_CAL_ACCEL not found
ERROR [param] Parameter SYS_CAL_BARO not found

nsh> param show SYS*
Symbols: x = used, + = saved, * = unsaved
x   SYS_FMU_TASK [338,829] : 0
```

**Comment by mcharleb** [comment]
@LorenzMeier I think moving PX4 to Vendor would make perfect sense. @dagar do you see any issue with that?

**Comment by dagar** [comment]
Organizing things uniformly sounds like the right approach, but where are the px4 boards going to end up once things are split into separate repos?

What I was aiming for with platforms is concentrating all NuttX pieces (including the nuttx board configurations) into a single directory which would then be broken out into another git repository. That NuttX-PX4 would effectively be a standalone project that builds a functioning NuttX on these boards. PX4-Firmware would simply be the flight software that runs on top.

So what's the next step? Where would all these pieces end up?

**Comment by mcharleb** [comment]
@dagar I think px4 should move to vendor it can be part of the main Firmware repo but ideally would not be. We could make a Vendor/misc and move the other boards to that, then Firmware is just flight control with posix support. if you want NuttX under PX4, then we need to allow platform support to live under a vendor directory which it currently doesn't support. You would currently have to aggregate the 3 repos when they are separated to build a PX4 HW target:
- Firmware
- Nuttx Platform support
- PX4 vendor 

What you are now proposing is having:
- Firmware
- PX4 + NuttX platform support

Snapdragon targets currently have:
- Firmware
- Qurt OS support
- ATLFlight vendor

That could be changed to:
- Firmware
- Qurt OS support + POSIX support additions (muorb) + ATLFlight

I just need clear objectives.

**Comment by mcharleb** [comment]
@davids5 The param problem is because the platform and vendor layers can now extend the base later (src), the param filtering limits the module dirs, but then it doesn't pick up the derived layer.

@dagar @lorenz The param filtering criteria is broken IMO. There is no constraint that a CMakeLists.txt file can't use files from another directory and that directory may contain a *params.c file.

I can include the base layers as well when looking for param files, but then you can never replace a param file in a layer, only add to it.

**Comment by dagar** [comment]
No, it's the same thing I was proposing on the call. A board requires particular platform support. Having it live within that platform/os directory keeps it simple and obvious.

Let's start with the boards that live within the open source project, looking at the pixhawk (nuttx_px4fmu-v2_default) as an example. The entire configuration for that board is found at Firmware/platforms/nuttx/boards/px4/fmu-v2. The pixhawk is a bit of a special case because it's actually a px4fmu-v2 + px4io-v2, however these are still separate boards (px4 builds) despite being one physical package. They must be built separately (cmake builds) and then combined.

	Firmware
	├── cmake
	├── Makefile
	├── platforms
	│   ├── nuttx
	│   │   ├── boards
	│   │   │   └── px4
	│   │   │       ├── fmu-v2
	│   │   │       │   ├── board_config.h
	│   │   │       │   ├── default.conf
	│   │   │       │   ├── nuttx-config
	│   │   │       │   ├── rc.sensors
	│   │   │       │   ├── src
	│   │   │       │   │   ├── px4fmu2_init.c
	│   │   │       │   │   ├── px4fmu_spi.c
	│   │   │       │   │   └── px4fmu_timer_config.c
	│   │   │       │   └── test.conf
	│   │   │       ├── fmu-v3
	│   │   │       │   └── rc.sensors
	│   │   │       ├── fmu-v4
	│   │   │       ├── fmu-v4pro
	│   │   │       ├── fmu-v5
	│   │   │       └── io-v2
	│   │   ├── CMakeLists.txt
	│   │   ├── drivers
	│   │   │   └── stm32
	│   │   └── NuttX
	│   │       ├── apps
	│   │       ├── CMakeLists.txt
	│   │       └── nuttx
	│   ├── posix
	│   │   └── boards
	│   │       ├── navio
	│   │       │   └── rpi
	│   │       │       ├── board_config.h
	│   │       │       ├── default.conf
	│   │       │       └── rc.sensors
	│   │       ├── parrot
	│   │       │   └── bebop
	│   │       └── px4
	│   │           └── sitl
	│   │               ├── board_config.h
	│   │               ├── default.conf
	│   │               └── rc.sensors
	│   └── qurt
	└── src
	    ├── drivers
	    └── modules

Now we want to think about how a 3rd party can extend this structure. Personally I think using repo to overlay each board into the appropriate spot is sufficient. For example if I were adding a new nuttx board I would place it at Firmware/platforms/nuttx/boards/vendor/xautopilot and need to aggregate...
 - PX4 Firmware
 - PX4 NuttX support (Firmware/platforms/nuttx)
 - vendor board (Firmware/platforms/nuttx/boards/vendor)

Commonly a vendor will want to take a board that's close to what they want, make a handful of changes and call it their own. Often I'll see people hack these boards in place because they don't know how to make a copy. With this they would simply copy the folder, rename, and go.

The top level vendor structure was suggested in addition to the core layout for even more flexibility, but I'm not sure it's needed. For instance what's the value in splitting out the PX4 board configs from PX4 NuttX? Those pieces aren't usable separately and every repository split is another layer of difficultly for a possible contributor and more overhead for maintainers. For a vendor wishing to keep their portion private the only downside to this is if you span multiple operating systems you'd need one repository for each. 

I don't really know the snapdragon, but if it lived in tree I believe it would be something like this.

	Firmware
	├── platforms
	│   ├── posix
	│   │   └── boards
	│   │       └── ATLFlight
	│   │           ├── cmake
	│   │           ├── eagle
	│   │           │   ├── board_config.h
	│   │           │   ├── default.conf
	│   │           │   ├── legacy.conf
	│   │           │   └── rc.sensors
	│   │           └── excelsior
	│   │               ├── board_config.h
	│   │               ├── default.conf
	│   │               ├── legacy.conf
	│   │               └── rc.sensors
	│   └── qurt
	│       └── boards
	│           └── ATLFlight
	│               ├── cmake
	│               ├── eagle
	│               │   ├── board_config.h
	│               │   ├── default.conf
	│               │   ├── legacy.conf
	│               │   ├── muorb.cmake
	│               │   ├── rc.sensors
	│               │   ├── test.cmake
	│               │   └── travis.cmake
	│               └── excelsior
	│                   ├── board_config.h
	│                   ├── default.conf
	│                   ├── legacy.conf
	│                   └── rc.sensors

One option for extending this out of tree is each platform ATLFlight becomes an external git repository.
Then for the full snapdragon you need to aggregate...
 - PX4 Firmware
 - PX4 QuRT support
 - ATLFlight posix
 - ATLFlight qurt

What I'm striving for after many hours of helping struggling developers, companies, students, hobbyists use and extend this platform is simplicity.

**Comment by mcharleb** [comment]
@dagar You have basically moved vendor under boards, and removed a way to have a single vendor repo. I can live with that if that makes it "easier", but then the other boards should be under boards/misc to simplify the hierarchy and make it uniform:  Firmware/${OS}/boards/${VENDOR}/${BOARD}

**Comment by mcharleb** [comment]
If we are all agreed on @dagar 's proposed layout I can redo the PR. The issue is still present with nuttx enhancements to what is in Firmware/src/* breaking the parameter scanning. 
Firmware/platforms/nuttx/src/modules/systemlib/CMakeLists.txt breaks the scanner because it won't find Firmware/src/modules/systemlib/system_params.c

**Comment by dagar** [comment]
I don't recall the motivation for vendor/board at that level, but I could go either way and I agree we don't need to support multiple conventions. The other boards all have vendors, although some don't currently have it as part of the name. Do you have a preference @davids5?

Regarding the hybrid architecture my issue with the proposed vendor structure is it lets you store the files in a single repository, but still doesn't solve the problem. At the PX4 "board" level we're talking about these are two separate boards. How do we orchestrate two separate cmake builds and tie them together as a single unit? I don't think that can be solved effectively within cmake. Maybe an optional Makefile include where you supply the hybrid target and dependencies on each PX4 build?

I'm still not opposed to an external overlay system (out of tree single vendor repo) as an extension to the external module mechanism we have now, I just think we need to first solve this as best we can for the open source project and lock down layout.

**Comment by davids5** [comment]
My preference is to keep everything regular. If it's vendor / board then each and every board should be pathed that way

**Comment by dagar** [comment]
Yes definitely regular, but do you have a preference for
platforms/nuttx/boards/VENDOR/BOARD or platforms/nuttx/boards/BOARD (BOARD
may include some portion of a vendor name).

**Comment by davids5** [comment]
I think platforms/nuttx/boards/VENDOR/BOAR wold be preferable.

**Comment by mcharleb** [comment]
@davids5  @dagar What do you want to use for ${VENDOR} for:

aerocore2  aerofc-v1  auav-x21  crazyflie  esc35-v1  mindpx-v2  px4  s2740vc-v1  tap-v1  zubaxgnss-v1

**Comment by davids5** [comment]
gumstix/aerocore2 intel/aerofc-v1 auav/x21 bitcraze/crazyflie auav/esc35-v1 airmind/mindpx-v2 px4 thiemar/s2740vc-v1 theautopilot/tap-v1 zubax/gnss-v1

**Comment by davids5** [comment]
@mcharleb - the Param culling was done in an effort to not use large amounts of Flash on boards that had 3 parameters. (The code would not fit) 

@dagar Do you see a way to do this and have the ability to replace for an added layer?

**Comment by mcharleb** [comment]
@dagar Are these vendor's correct? What is vendor for ocpoc?
parrot/bebop 
??/ocpoc 
rpi/rpi 
px4/sitl

**Comment by mcharleb** [comment]
@davids5 @dagar Would you be open to adding a variable in the CMakeLists.txt file to specify the parameters file(s) for that module? Then the culling becomes much simpler to support with overlay extensions. Ideally they's just get added to a global scope variable.
For platforms like snapdragon that need to keep the parameters for 2 different builds in sync, we may need a way to pass in a config file that lists the shared modules to use for generating the parameters

**Comment by TSC21** [comment]
OcPoC is Aerotenna - https://aerotenna.com/ocpoc-cyclone/

**Comment by jywilson** [comment]
@mcharleb Just a question on the semantics... 

Is the term vendor really intended to refer to the manufacturer of the board?  If so, would it be better to use the term "oem"?  Also, if a particular board is being cloned and does not have an identifiable manufacturer perhaps you would just use a default name indicating this, such as "vendor" (or oem, as described above).

Overall I like the direction this taking.  This will allow PX4 to scale to many different boards while allowing the flight stack to evolve independently.

**Comment by mcharleb** [comment]
@jywilson I don't like the name vendor either, but I think I like OEM less. Its Company/Org put I don't have a good term for that.

**Comment by mcharleb** [comment]
@TSC21 Thanks for the Aerotenna info.

**Comment by dagar** [comment]
"Raspiberry pi" is actually Emlid Navio2. https://emlid.com/navio/

**Pull request closed without merging**